### PR TITLE
Code restructured to a package

### LINKED ISSUE #45: Restructure the code to a package structure

### PR DESCRIPTION
**Pull request by krzmip**
No change in functionality (equal to current develop), but fundamental change in file structure.
Closes #45 

Notes:
1. R package names do not accept underscores (only alphanumeric and dots), so the package name is "climate.narrative" (but the main function name is still climate_narrative with an underscore)
2. The app can be run in two ways now (both are independent of R working directory):
    - install the package using devtools::install(path to the folder) and then run climate_narrative()
    - work without installation: devtools::load_all() and then run climate_narrative()
3. the devtools::check() command shows no errors and two warnings:
    - some functions are not documented
    - some filenames are not fully portably (probably it is due to spaces in their names)
4. the package function documentation (where available) can be accessed in R in the regular way (e.g. ?climate_narrative)

### CODE REVIEW

File: R/server.R [hunk 1]
```diff
@@ -110,12 +110,22 @@ server <- function(input, output, session) {
     if (!exists("temp_md_scenario")) temp_md_scenario <- tempfile(fileext = ".md")
     file_conn <- file(temp_md_scenario)
     scenario_no <- c(
-      which(sapply(scenarios, `[[`, i = "name") == report_selection),
+      which(sapply(global$scenarios, `[[`, i = "name") == report_selection),
       length(report_contents()) - 1
     )
+    tempfun=function(x) gsub("\\(([[:graph:]]*)(.png)", paste0("(", getwd(),"/www/", "\\1\\2"), x,perl=T)
+    #xmpl="![NGFS scenarios Framework](NGFS_scenarios_Framework_Orderly.png)"
```
Comment (JohnAdders) [inline R/server.R:117]: no commented code

File: R/server.R [hunk 2]
```diff
@@ -110,12 +110,22 @@ server <- function(input, output, session) {
...
+    #print(xmpl)
+    #print(tempfun(xmpl))
     writeLines(
       # plus one is for the title, not included in 'scenarios' but included in 'report_contents'
-      report_contents()[c(1 + scenario_no)],
+      tempfun(report_contents()[c(1 + scenario_no)]),
+      # gsub(
```
Comment (JohnAdders) [inline R/server.R:123]: no commented code

File: R/server.R [hunk 3]
```diff
@@ -110,12 +110,22 @@ server <- function(input, output, session) {
     if (!exists("temp_md_scenario")) temp_md_scenario <- tempfile(fileext = ".md")
     file_conn <- file(temp_md_scenario)
     scenario_no <- c(
-      which(sapply(scenarios, `[[`, i = "name") == report_selection),
+      which(sapply(global$scenarios, `[[`, i = "name") == report_selection),
       length(report_contents()) - 1
     )
+    tempfun=function(x) gsub("\\(([[:graph:]]*)(.png)", paste0("(", getwd(),"/www/", "\\1\\2"), x,perl=T)
```
Comment (JohnAdders) [inline R/server.R:116]: better name for function, also prefer <-

File: R/server.R (hunk 1 above)
Comment (krzmip) [reply R/server.R]: addressed

File: R/server.R (hunk 2 above)
Comment (krzmip) [reply R/server.R]: addressed

File: R/server.R (hunk 3 above)
Comment (krzmip) [reply R/server.R]: addressed

**Pull request merged**